### ISSUE TITLE
Load more than one custom dll possible?

### ISSUE THREAD

**Issue by jsc-paneda** [taviso/loadlibrary#76]
I have successfully used loadlibrary to load a custom dll and used some of its functions.

However one of the dll functions has a dependency to the GetVersionExA in the kernel32.dll.

So I used ['detours'](https://github.com/microsoft/Detours) to create a myhook.dll that replaces this function. This worked fine on Windows. However on Linux side I realized that I pretty much only replaced the kernel32.dll dependency with myhook.dll dependency. And so the issue remains.

If only I could somehow load myhook.dll also? Or is there some other way? Or do I need Wine? Its all a bit of black magic right now..

**Comment by cube0x8** [comment]
Umh... I know I am commenting something quite old, but [here](https://github.com/taviso/loadlibrary/blob/9910ceda2e48955e72ab523a57a9aff75b125e47/peloader/winapi/Version.c#L110) you can find the hook for the API you need.

So, if your DLL has only that dependency, I guess you should be fine.

**Comment by jsc-paneda** [comment]
Yes a bit old, trying to remember myself..but I think the question was meant as a general question regarding possibility to load multiple dll:s, i.e. 'yes' or 'no'. As I recall my conclusion was 'no' and would just like to get confirmation on that.

**Comment by cube0x8** [comment]
I am quite sure that the short answer to your question is: no.

i.e. if your .dll calls some of the kernel32.dll APIs, you cannot use loadlibrary to load your .dll along with kernel32.dll to make your dll able to use the exported functions of kernel32.dll.

Anyway, keep in mind that this does not mean you cannot load your dll at all if it calls third-party APIs. You can still load it, using the loadlibrary's hooks.

I hope I explained myself better this time.

**Comment by jsc-paneda** [comment]
Understood and as I expected. Thanks, btw I'm using loadlibrary in my app successfully so I'm very pleased with your time and effort to put this together!
